### PR TITLE
Centos7 python 3.7

### PR DESCRIPTION
**Pull request by sam-github**
Installs python3 in /opt/python-3.7/bin, as python3:
    
            % /opt/python-3.7/bin/python3 --version
            Python 3.7.3

### CODE REVIEW

**Comment by sam-github** [comment]
@mhdawson LGTY?

@rvagg I don't know state of python3 on ARM, if you need it build because packages don't exist, this will do it.

/to @cclauss @nodejs/python 

Also, I'm not sure who is working on updating python support, but note carefully the location of the python3. I'm not sure, is it unreasonably conservative? I could set the prefix to /usr/local to have it in the default path, but that would force there to be only one python3 on the system -- maybe that's OK, even desireable?

**Comment by sam-github** [comment]
FTR:
```
build/ansible (centos7-python-3.7 % u=) % ssh test-osuosl-centos7-ppc64_le-1 /opt/python-3.7/bin/python3 --version                                                           
Python 3.7.3
build/ansible (centos7-python-3.7 % u=) % ssh test-osuosl-centos7-ppc64_le-2 /opt/python-3.7/bin/python3 --version
bash: /opt/python-3.7/bin/python3: No such file or directory
build/ansible (centos7-python-3.7 % u=) % ssh test-osuosl-centos7-ppc64_le-1 ls /opt/python-3.7/bin
2to3
2to3-3.7
idle3
idle3.7
pydoc3
pydoc3.7
python3
python3.7
python3.7-config
python3.7m
python3.7m-config
python3-config
pyvenv
pyvenv-3.7
```

Once we decide on where python should be, I can update this PR, and apply it to the second centos7-ppc machine.

**Comment by cclauss** [comment]
This is great work!  Thanks for doing it.  I would prefer if the path to ___python2__ and the path to __python3__ should be the same.  That is how it is done on most current distros (if they still ship legacy Python).

**Comment by sam-github** [comment]
They can't literally be the same paths, I don't think compiling something and installing it into `/usr` is a good idea, specifically because that is where distros put packages. I can put it in `/usr/local`, though, it will be in the default PATH, and its where locally installed packages usually go. I'll redo this PR.

**Comment by rvagg** [comment]
on arm64 centos7 python34.aarch64 and python36.aarch64 are available, if we need 37 then I guess we need to go this route too

File: ansible/roles/baselayout/tasks/partials/repo/centos7.yml [hunk 1]
```diff
@@ -39,3 +39,43 @@
...
+  when: "arch == 'ppc64'"
+  unarchive:
+    src: /tmp/Python-3.7.3.tgz
+    remote_src: yes
+    dest: /tmp/
+  when: build_python37.stat.exists == False
+
+- name: centos7 | ppc64 | configure python 3.7
+  when: "arch == 'ppc64'"
+  shell: /tmp/Python-3.7.3/configure --prefix=/opt/python-3.7
```
Comment (cclauss) [inline ansible/roles/baselayout/tasks/partials/repo/centos7.yml]: I thought this was going to be __usr/local__

**Comment by sam-github** [comment]
I think this is landable, can I get a re-approval, please?

@mhdawson you'll need to do the release machine, see below.

```
build/ansible (centos7-python-3.7 *% u=) % parallel-ssh -h ../hosts/osuosl-centos7-ppc64_le -i python3 -m pip --version                                                      
[1] 13:42:43 [FAILURE] release-osuosl-centos7-ppc64_le-1 Exited with error code 255
Stderr: ssh: Could not resolve hostname release-osuosl-centos7-ppc64_le-1: Temporary failure in name resolution                                                              
[2] 13:42:44 [SUCCESS] test-osuosl-centos7-ppc64_le-2
pip 19.0.3 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
[3] 13:42:45 [SUCCESS] test-osuosl-centos7-ppc64_le-1
pip 19.0.3 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
build/ansible (centos7-python-3.7 *% u=) % parallel-ssh -h ../hosts/osuosl-centos7-ppc64_le -i python3 --version       
[1] 13:42:54 [FAILURE] release-osuosl-centos7-ppc64_le-1 Exited with error code 255
Stderr: ssh: Could not resolve hostname release-osuosl-centos7-ppc64_le-1: Temporary failure in name resolution
[2] 13:42:55 [SUCCESS] test-osuosl-centos7-ppc64_le-1
Python 3.7.3
[3] 13:42:55 [SUCCESS] test-osuosl-centos7-ppc64_le-2
Python 3.7.3
build/ansible (centos7-python-3.7 *% u=) % parallel-ssh -h ../hosts/osuosl-centos7-ppc64_le -i which python3    
[1] 13:43:03 [FAILURE] release-osuosl-centos7-ppc64_le-1 Exited with error code 255
Stderr: ssh: Could not resolve hostname release-osuosl-centos7-ppc64_le-1: Temporary failure in name resolution
[2] 13:43:04 [SUCCESS] test-osuosl-centos7-ppc64_le-1
/usr/local/bin/python3
[3] 13:43:04 [SUCCESS] test-osuosl-centos7-ppc64_le-2
/usr/local/bin/python3
```

**Comment by cclauss** [comment]
What is next step here?

**Comment by sam-github** [comment]
Applying the change to the release machine, and merging.

**Comment by sam-github** [comment]
Landed in 617b2b652682f, 4618d4864cea5edc

**Pull request merged**